File: app/src/main/java/com/amirreza/quizapplication/feature_quiz/presentation/fragment_history/HistoryFragment.java
package com.amirreza.quizapplication.feature_quiz.presentation.fragment_history;

import android.os.Bundle;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.fragment.app.Fragment;
import androidx.navigation.Navigation;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import com.amirreza.quizapplication.util.base.QuizBaseFragment;
import com.amirreza.quizeapplication.R;
import com.amirreza.quizeapplication.databinding.FragmentHistoryBinding;
import com.amirreza.quizeapplication.databinding.FragmentHomeBinding;


public class HistoryFragment extends QuizBaseFragment {
    private FragmentHistoryBinding binding;

    @Override
    public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
        binding = FragmentHistoryBinding.inflate(inflater,container,false);
        return binding.getRoot();
    }

    @Override
    public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceState) {
        super.onViewCreated(view, savedInstanceState);

    }

}

File: app/src/main/java/com/amirreza/quizapplication/feature_quiz/presentation/fragment_history/HistoryFragment.kt
package com.amirreza.quizapplication.feature_quiz.presentation.fragment_history

import com.amirreza.quizapplication.util.base.QuizBaseFragment
import android.view.LayoutInflater
import android.view.ViewGroup
import android.os.Bundle
import android.view.View
import androidx.recyclerview.widget.LinearLayoutManager
import androidx.recyclerview.widget.RecyclerView
import com.amirreza.quizeapplication.databinding.FragmentHistoryBinding
import org.koin.android.ext.android.inject

class HistoryFragment : QuizBaseFragment() {
    private lateinit var binding: FragmentHistoryBinding
    private val viewModel:HistoryFragmentViewModel by inject()

    override fun onCreateView(
        inflater: LayoutInflater,
        container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View {
        binding = FragmentHistoryBinding.inflate(inflater, container, false)
        return binding.root
    }

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)

        val layoutManager = LinearLayoutManager(requireContext(), RecyclerView.VERTICAL,false)
        val recyclerView = binding.quizHistoryRecyclerView
        recyclerView.layoutManager = layoutManager

        viewModel.examHistory.observe(viewLifecycleOwner){
            setEmptyVisibility(it.size==0)
            recyclerView.adapter = QuizHistoryAdapter(it)
        }
    }

    fun setEmptyVisibility(isVisible:Boolean){
        binding.emptyState.visibility = if(isVisible) View.VISIBLE else View.GONE
    }
}